### PR TITLE
PR associations

### PR DESCRIPTION
**Pull request by DarioAlessioR**
PR for branch associations, for activity OOP school library: set up associations.

This branch includes the associations between the different classes, according to the following project requirements:

- between Classroom and Student
- between Person and Book using the intermediate class Rental

The implemented features were tested in the main.rb file.

Thank you in advance for your review.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
```
Comment (tufoinnkuo10) [inline book.rb]: - [ ] please the method for addint rental should take in the person and date. the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association
you can use something like this

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: person.rb [hunk 2]
```diff
@@ -20,6 +26,11 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
```
Comment (tufoinnkuo10) [inline person.rb]: - [ ] Great job. here kindly ensure the add rental method takes in the book and date. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. something like 👇🏾 

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @DarioAlessioR 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ associations created correctly
✅ all classes available

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 2 above)
Comment (DarioAlessioR) [reply person.rb]: Changes done. Thank you for the suggestions @tufoinnkuo10 .

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @DarioAlessioR 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**